### DISCUSSION TITLE
Dockerize this service

### DISCUSSION THREAD

**Discussion by Vivek230800** [EmulatorJS/EmulatorJS#309] [Q&A]
Could anyone please guide me about how can I dockerize this service with nginx? I can't find a way. I want to run a different container as well. And bcoz if that emulatorjs is not working.

**Comment by ethanaobrien** [comment]
I’m not familiar with docker. Hopefully someone else can help.

Would you mind linking to me on some references on what it is and maybe I can look into this?

**Reply by Vivek230800** [reply]
Sure. Docker works like this. This image is the best explanation.![docker-containerized-appliction-blue-border_2.png](https://user-images.githubusercontent.com/59607359/183504016-4f412a5c-357e-4211-9376-0b6289b69dca.png)

**Reply by Vivek230800** [reply]
So if one service is using resources which overload the ram then the whole os stops responding. Docker is used to prevent that and run multiple services while we can also check their status.

**Reply by ethanaobrien** [reply]
I'll take a look, not sure when I can do so or if I'll be able to figure it out. Hopefully someone else can also help on this.

**Reply by allancoding** [reply]
I have done some docker images and containers. I will see if I can make a image when I finish with what I am working on. I would be thinking of a nginx server with php and we could also add emulatorjs netplay.

**Comment by silvavn** [comment]
If the maintainers are ok with it, I can open a PR with docker.

**Reply by ethanaobrien** [reply]
If you take a look at the faq (https://emulatorjs.org/docs/FAQ.html#q-why-is-the-demo-the-only-official-emulatorjs-instance) you can see the approach I've taken on this. A docker server isnt really the aim of this project, but there are branches of this project that I promote that are "docker versions" of this project
